packages/@aws-cdk/integ-tests/README.md

```diff
@@ -428,3 +428,52 @@ integ.assertions.awsApiCall('S3', 'putObject', {
 }));
 ```
 
+#### Wait for results
+
+A common use case when performing assertions is to wait for a condition to pass. Sometimes the thing
+that you are asserting against is not done provisioning by the time the assertion runs. In these
+cases it is possible to run the assertion asynchronously by calling the `wait()` method.
+
+Taking the example above of executing a StepFunctions state machine, depending on the complexity of
+the state machine, it might take a while for it to complete.
+
+```ts
+declare const app: App;
+declare const stack: Stack;
+declare const sm: IStateMachine;
+
+const testCase = new IntegTest(app, 'IntegTest', {
+  testCases: [stack],
+});
+
+// Start an execution
+const start = testCase.assertions.awsApiCall('StepFunctions', 'startExecution', {
+  stateMachineArn: sm.stateMachineArn,
+});
+
+// describe the results of the execution
+const describe = testCase.assertions.awsApiCall('StepFunctions', 'describeExecution', {
+  executionArn: start.getAttString('executionArn'),
+}).expect(ExpectedResult.objectLike({
+  status: 'SUCCEEDED',
+})).waitForAssertions();
+```
+
+When you call `wait()` the assertion provider will continuously make the `awsApiCall` until the
```

Comment:
[nit] `wait` as a full word match is not mentioned in the code snippet.

Reply:
[check] The logic operates at a low enough rate that throttling will not be an influential factor, right?

Reply:
> [nit] wait as a full word match is not mentioned in the code snippet.

Good catch! forgot to update.

> [check] The logic operates at a low enough rate that throttling will not be an influential factor, right?

I would assume so, but if not the user can update the parameters to make the call less frequently